src/odemis/gui/cont/tabs.py

```diff
@@ -910,11 +911,24 @@ def __init__(self, name, button, panel, main_frame, main_data):
              {"name": "Spectrum",
               "stream_classes": SpectrumStream,
               }),
-            (viewports[3],
-             {"name": "Monochromator",
-              "stream_classes": MonochromatorSettingsStream,
-              }),
         ])
+
+        # depending on HW choose which viewport should be connected to a stream
+        # TODO: for now only time correlator HW or streak cam HW handled
+        if main_data.streak_ccd:
+            vpv[viewports[3]] = {
+                "name": "Temporal Spectrum",
+                "stream_classes": TemporalSpectrumStream,
+            }
+
+        else:
+            vpv[viewports[4]] = {
+                "name": "Monochromator",
+                "stream_classes": MonochromatorSettingsStream,
+            }
+
+        # TODO what if none of both? double one of the other viewports? Can we go with 3?
```

Comment:
There should always be 4 viewports (or more).
If none of the the 2 HW are available, it's the same as currently, we display the monochromator. If both are available... we need to be more clever, but for now that doesn't exists. So let's just put a TODO, and/or log a warning  inside the `if main_data.streak_ccd` when `main_data.monochromator` exists.